gdpr/gdpr.go

```diff
@@ -65,3 +67,17 @@ type ErrorMalformedConsent struct {
 func (e *ErrorMalformedConsent) Error() string {
 	return "malformed consent string " + e.consent + ": " + e.cause.Error()
 }
+
+// SignalParse parses a raw signal and returns
+func SignalParse(rawSignal string) (Signal, error) {
+	i, err := strconv.Atoi(rawSignal)
+
+	if err != nil {
+		return SignalAmbiguous, err
+	}
+	if i != 0 && i != 1 {
+		return SignalAmbiguous, &errortypes.BadInput{"GDPR signal should be integer 0 or 1"}
```

Comment:
Linters will warn you against using positional/unkeyed values. Please add the `Message:` key in the initializer for the message content.

Reply:
Good catch. Not sure how I missed that one.